s3-commons/src/main/java/io/aiven/kafka/connect/s3/S3SourceBaseConfig.java

```diff
@@ -0,0 +1,449 @@
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigException;
+
+import io.aiven.kafka.connect.common.config.SourceCommonConfig;
+import io.aiven.kafka.connect.common.config.validators.FileCompressionTypeValidator;
+import io.aiven.kafka.connect.common.config.validators.NonEmptyPassword;
+import io.aiven.kafka.connect.common.config.validators.OutputFieldsValidator;
+import io.aiven.kafka.connect.common.config.validators.UrlValidator;
+import io.aiven.kafka.connect.iam.AwsStsEndpointConfig;
+import io.aiven.kafka.connect.iam.AwsStsRole;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.internal.BucketNameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3SourceBaseConfig extends SourceCommonConfig {
+    public static final Logger LOGGER = LoggerFactory.getLogger(S3SourceBaseConfig.class);
+
+    @Deprecated
+    public static final String OUTPUT_COMPRESSION = "output_compression";
+    @Deprecated
+    public static final String OUTPUT_COMPRESSION_TYPE_GZIP = "gzip";
+    @Deprecated
+    public static final String OUTPUT_COMPRESSION_TYPE_NONE = "none";
+
+    @Deprecated
+    public static final String OUTPUT_FIELDS = "output_fields";
+    @Deprecated
+    public static final String TIMESTAMP_TIMEZONE = "timestamp.timezone";
+    @Deprecated
+    public static final String TIMESTAMP_SOURCE = "timestamp.source";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_KEY = "key";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_OFFSET = "offset";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_TIMESTAMP = "timestamp";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_VALUE = "value";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_HEADERS = "headers";
+
+    @Deprecated
+    public static final String AWS_ACCESS_KEY_ID = "aws_access_key_id";
+    @Deprecated
+    public static final String AWS_SECRET_ACCESS_KEY = "aws_secret_access_key";
+    @Deprecated
+    public static final String AWS_S3_BUCKET = "aws_s3_bucket";
+    @Deprecated
+    public static final String AWS_S3_ENDPOINT = "aws_s3_endpoint";
+    @Deprecated
+    public static final String AWS_S3_REGION = "aws_s3_region";
+    @Deprecated
+    public static final String AWS_S3_PREFIX = "aws_s3_prefix";
+
+    // FIXME since we support so far both old style and new style of property names
+    // Importance was set to medium,
+    // as soon we will migrate to new values it must be set to HIGH
+    // same for default value
+    public static final String AWS_ACCESS_KEY_ID_CONFIG = "aws.access.key.id";
```

Comment:
these kind of configs can be moved to common config ? now they are duplicated, and if a change needs to happen, it should be done in both places.